src/useScrollSync.ts

```diff
@@ -0,0 +1,91 @@
+import type React from "react";
+import { useEffect } from "react";
+
+interface Options {
+  proportional?: boolean;
+  axis?: "horizontal" | "vertical" | "both";
+}
+
+export const useScrollSync = (
+  refs: React.MutableRefObject<HTMLElement | null>[],
+  { proportional = true, axis = "both" }: Options = {}
+): void => {
+  useEffect(() => {
+    const nodes = refs
+      .map((ref) => ref.current)
+      .filter((n) => n !== null) as HTMLElement[];
```

Comment:
One loop should be enough here. No need for this map. Just access the `.current` directly.

Reply:
I actually have a local implementation that makes the ref handling much simpler and should address this. Namely, the `useScrollSync` hook will just accept a list of `HTMLElement` (rather than a list of refs to `HTMLElement`s).

I'll update with this change soon.

Reply:
I've updated the PR to include this new implementation. For now just adding as another commit, but if we get to the point where we want to merge it, I'm happy to squash/rebase commits as needed.